Data/Sequence.hs

```diff
@@ -13,34 +16,100 @@
 -- Maintainer  :  libraries@haskell.org
 -- Portability :  portable
 --
--- General purpose finite sequences.
--- Apart from being finite and having strict operations, sequences
--- also differ from lists in supporting a wider variety of operations
--- efficiently.
+-- = Finite sequences
+--
+-- The @'Seq' a@ type represents a finite sequence of values of
+-- type @a@.
+--
+-- Sequences generally behave very much like lists.
+--
+-- * The class instances for sequences are all based very closely on those for
+-- lists.
+--
+-- * Many functions in this module have the same names as functions in
+-- the "Prelude" or in "Data.List". In almost all cases, these functions
+-- behave analogously. For example, 'filter' filters a list in exactly the
```

Comment:
filters a _sequence_ in exactly the same way that 'Prelude.filter' filters a _list_.

Reply:
Thanks.